modules/externalipam/ipam_validator.rb

```diff
@@ -0,0 +1,49 @@
+require 'proxy/validations'
+require 'externalipam/externalipam'
+require 'resolv'
+
+# Module containing validation methods for use by all External IPAM provider implementations
+module Proxy::Ipam::IpamValidator
+  include ::Proxy::Validations
+  include Proxy::Ipam::IpamHelper
+
+  def validate_required_params!(required_params, params)
+    err = []
+    required_params.each do |param|
+      unless params[param.to_sym]
+        err.push errors[param.to_sym]
+      end
+    end
+    raise Proxy::Validations::Error, err unless err.empty?
+  end
+
+  def validate_ip!(ip)
```

Comment:
We already have one:
https://github.com/theforeman/smart-proxy/blob/6b75029e01767cad7873a5c58971fbbde5aabb65/lib/proxy/validations.rb#L33-L41